jot/JotTouchBezier.m

```diff
@@ -35,40 +42,58 @@ - (void)dealloc
 	CGPathRelease(_path);
 }
 
-- (void)jotDraw
+- (void)jotDrawWithScaling:(BOOL)shouldScale
 {
     if (self.constantWidth) {
-		[self drawStrategy1];
+		[self drawStrategy1WithScaling:shouldScale];
     } else {
-		[self drawStrategy2];
+		[self drawStrategy2WithScaling:shouldScale];
     }
 }
 
+- (CGMutablePathRef)scaledPath {
+    CGAffineTransform scaleTransform = CGAffineTransformMakeScale(self.outputScaleFactor, self.outputScaleFactor);
+    return CGPathCreateMutableCopyByTransformingPath(self.path, &scaleTransform);
+}
+
 - (void)generatePath {
 	if (_path) CGPathRelease(_path);
 	_path = CGPathCreateMutable();
 	CGPathMoveToPoint(_path, NULL, self.startPoint.x, self.startPoint.y);
 	CGPathAddCurveToPoint(_path, NULL, self.controlPoint1.x, self.controlPoint1.y, self.controlPoint2.x, self.controlPoint2.y, self.endPoint.x, self.endPoint.y);
 }
 
-- (void)drawStrategy1 {
+- (void)drawStrategy1WithScaling:(BOOL)shouldScale {
 	CGContextRef context = UIGraphicsGetCurrentContext();
 	if (!context) {
 		return;
 	}
-	CGContextAddPath(context, _path);
+    CGMutablePathRef path = shouldScale ? self.scaledPath : _path;
+	CGContextAddPath(context, path);
 	CGContextSetLineWidth(context, self.startWidth);
 	CGContextSetLineCap(context, kCGLineCapRound);
 	[self.strokeColor setStroke];
 	CGContextStrokePath(context);
 }
 
-- (void)drawStrategy2 {
+- (void)drawStrategy2WithScaling:(BOOL)shouldScale {
 	CGContextRef context = UIGraphicsGetCurrentContext();
 	if (!context) {
 		return;
 	}
-	CGFloat widthDelta = self.endWidth - self.startWidth;
+    
+    CGFloat scaleFactor = shouldScale ? self.outputScaleFactor : 1.f;
+    CGPoint scaledStartPoint = CGPointMake(self.startPoint.x * scaleFactor,
+                                           self.startPoint.y * scaleFactor);
+    CGPoint scaledEndPoint = CGPointMake(self.endPoint.x * scaleFactor,
+                                         self.endPoint.y * scaleFactor);
+    CGPoint scaledControlPoint1 = CGPointMake(self.controlPoint1.x * scaleFactor,
+                                              self.controlPoint1.y * scaleFactor);
+    CGPoint scaledControlPoint2 = CGPointMake(self.controlPoint2.x * scaleFactor,
+                                              self.controlPoint2.y * scaleFactor);
+    CGFloat scaledStartWidth = self.startWidth * scaleFactor;
+    CGFloat scaledEndWidth = self.endWidth * scaleFactor;
+	CGFloat widthDelta = scaledEndWidth - scaledStartWidth;
```

Comment:
Couldn't you apply a scale transform to the context and avoid a lot of this math?